src/xSTIR/cSTIR/cstir_p.cpp

```diff
@@ -433,6 +433,67 @@ sirf::cSTIR_setQuadraticPriorParameter
 		objectFromHandle<xSTIR_QuadraticPrior3DF>(hp);
 	if (sirf::iequals(name, "only_2D"))
 		prior.only2D(dataFromHandle<int>((void*)hv));
+	else if (sirf::iequals(name, "kappa")) {
+		STIRImageData& id = objectFromHandle<STIRImageData>(hv);
+		prior.set_kappa_sptr(id.data_sptr());
+	}
+	else
+		return parameterNotFound(name, __FILE__, __LINE__);
+	return new DataHandle;
+}
+
+void*
+sirf::cSTIR_QuadraticPriorParameter
+(DataHandle* hp, const char* name)
+{
+	xSTIR_QuadraticPrior3DF& prior =
+		objectFromHandle<xSTIR_QuadraticPrior3DF >(hp);
+	if (sirf::iequals(name, "kappa")) {
+		auto sptr_im = prior.get_kappa_sptr();
+		auto sptr_id = std::make_shared<STIRImageData>(*sptr_im);
```

Comment:
this implies we're making a copy for no real reason. @evgueni-ovtchinnikov is there  a way to avoid this? i.e. construct a `STIRImageData` from a `shared_ptr<DiscretisedDensity<3,float>>`?

Reply:
I have no idea why Richard decided to make a copy here (and in similar places). I would just return `newObjectHandle(sptr_im)`.